src/grpcClient.ts

```diff
@@ -148,69 +139,57 @@ export class TraderClient implements TraderClientInterface {
     Array<{ baseAsset: string; quoteAsset: string; feeBasisPoint: number }>
   > {
     return new Promise((resolve, reject) => {
-      this.client.markets(new messages.MarketsRequest(), (err, response) => {
-        if (err) return reject(err);
-        const list = response!
-          .getMarketsList()
-          .map((mktWithFee: types.MarketWithFee) => ({
-            baseAsset: mktWithFee!.getMarket()!.getBaseAsset(),
-            quoteAsset: mktWithFee!.getMarket()!.getQuoteAsset(),
-            feeBasisPoint: mktWithFee!.getFee()!.getBasisPoint(),
-          }));
-        resolve(list);
-      });
+      this.client.listMarkets(
+        messages.ListMarketsRequest.create(),
+        (err, response) => {
+          if (err) return reject(err);
+          const list = response!.markets.map(
+            (mktWithFee: types.MarketWithFee) => ({
+              baseAsset: mktWithFee!.market!.baseAsset,
+              quoteAsset: mktWithFee!.market!.quoteAsset,
+              feeBasisPoint: Number(mktWithFee!.fee!.basisPoint),
+            })
+          );
+          resolve(list);
+        }
+      );
     });
   }
 
   marketPrice(
```

Comment:
maybe better to rename this `preview` or something like that?

Reply:
will break apps tho